src/acc/libsmm_acc/libcusmm/tune.py

```diff
@@ -42,6 +42,7 @@ def main():
     param_fn = options.params
     assert param_fn in arch_number.keys(), "Cannot find compute version for file " + param_fn
     arch = arch_number[param_fn]
+    gpu_properties = json.load(open('kernels/gpu_properties.json'))["sm_" + str(arch)]
```

Comment:
I would recommend something like:

```python
with open('kernels/gpu_properties.json') as fhandle:
    gpu_properties = json.load(fhandle)["sm_{}".format(arch)]
```

to make sure we close that file again at a defined point.

Reply:
good point!